CMakeLists.txt

```diff
@@ -523,6 +523,62 @@ if(CIRCT_BINDINGS_TCL_ENABLED)
   message(STATUS "Found TCL executable: ${TCL_TCLSH}")
 endif()
 
+#-------------------------------------------------------------------------------
+# slang Verilog Frontend
+#-------------------------------------------------------------------------------
+
+option(CIRCT_SLANG_FRONTEND_ENABLED "Enables the slang Verilog frontend." OFF)
+option(CIRCT_SLANG_BUILD_FROM_SOURCE
+  "Build slang from source instead of finding an installed package" ON)
+
+llvm_canonicalize_cmake_booleans(CIRCT_SLANG_FRONTEND_ENABLED)
+llvm_canonicalize_cmake_booleans(CIRCT_SLANG_BUILD_FROM_SOURCE)
+
+if(CIRCT_SLANG_FRONTEND_ENABLED)
+  message(STATUS "slang Verilog frontend is enabled")
+  if(CIRCT_SLANG_BUILD_FROM_SOURCE)
+    # Build slang as part of CIRCT (see https://sv-lang.com/building.html)
+    message(STATUS "Building slang from source")
+    include(FetchContent)
+    FetchContent_Declare(
+      slang
+      GIT_REPOSITORY https://github.com/MikePopoloski/slang.git
+      GIT_TAG v3.0
+      GIT_SHALLOW ON)
+    set(FETCHCONTENT_TRY_FIND_PACKAGE_MODE "NEVER")
+
+    # Force Slang to be built as a static library to avoid messing around with
+    # RPATHs and installing a slang dylib alongside CIRCT. The static library
+    # will embed Slang into ImportVerilog.
+    set(original_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+    set(original_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
```

Comment:
We'd still have to save the originals, since you'd want to restore the CIRCT CXX flags for the `ImportVerilog` build itself. My reasoning for putting this here was to have Slang the dependency be prominently visible in the main CMakeLists.txt file, instead of hidden away in a tool, and then instead only build that tool if the dependency has been found. You also want to enable/disable tests based on whether Slang is on and has been found. Burying that in a library feels like an antipattern.